interp/builtin.go

```diff
@@ -984,14 +984,16 @@ func (r *Runner) readLine(ctx context.Context, raw bool) ([]byte, error) {
 	}
 }
 
-func (r *Runner) changeDir(ctx context.Context, path string) uint8 {
+func (r *Runner) changeDir(ctx context.Context, cmd, path string) uint8 {
 	path = cmp.Or(path, ".")
 	path = r.absPath(path)
 	info, err := r.stat(ctx, path)
 	if err != nil || !info.IsDir() {
+		r.errf("%s: the directory '%s' does not exist\n", cmd, path)
```

Comment:
use `%q` for the path, here and in the line below - that way we get consistently good quoting. it doesn't matter that we're not like bash - the tests already don't expect that because of `#JUSTERR`